File: app/src/jvx/java/art/coded/wireframe/viewmodel/DetailViewModel.java
package art.coded.wireframe.viewmodel;

import android.content.Context;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import art.coded.wireframe.model.entity.Element;
import art.coded.wireframe.model.ElementRepository;

import java.util.List;

public class DetailViewModel extends ViewModel {

    private static final String LOG_TAG = DetailViewModel.class.getSimpleName();

    private LiveData<List<Element>> mData;
    private ElementRepository mRepository;

    public void loadData(Context context) {
        mRepository = new ElementRepository(context);
        mData = mRepository.getAllElements();
    }

    public LiveData<List<Element>> getData() { return mData; }
    public void addData(Element element) { mRepository.insert(element); }
    public void removeAllData() { mRepository.deleteAll(); }
}

File: app/src/ktx/java/art/coded/wireframe/viewmodel/DetailViewModel.kt
package art.coded.wireframe.viewmodel

import androidx.lifecycle.*
import art.coded.wireframe.model.ElementRepository
import art.coded.wireframe.model.entity.Element
import kotlinx.coroutines.launch

private val LOG_TAG = DetailViewModel::class.java.simpleName

class DetailViewModel(private val repository: ElementRepository) : ViewModel() {

    val actionPending = MutableLiveData<Boolean>()

    fun addData(element: Element) {
        actionPending.postValue(true)
        viewModelScope.launch {
            repository.insert(element)
        }
        actionPending.postValue(false)
    }

    fun removeAllData() {
        actionPending.postValue(true)
        viewModelScope.launch {
            repository.deleteAll()
        }
        actionPending.postValue(false)
    }
}

class DetailViewModelFactory(private val repository: ElementRepository) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return DetailViewModel(repository) as T
    }
}